### ISSUE TITLE
Junit reporter improvements

### ISSUE THREAD

**Issue by rockandska** [inspec/inspec#3006]
Hi Inspec Team,

### 🎛 Description

🙋 feature request

Not sure if there is an official Junit XSD but the attributes for "testsuite(s)"/"testcase" elements used in most cases seems to be the bellow ones : 

```
    <xs:element name="testcase">
        <xs:complexType>
            <xs:sequence>
                <xs:element ref="skipped" minOccurs="0" maxOccurs="1"/>
                <xs:element ref="error" minOccurs="0" maxOccurs="unbounded"/>
                <xs:element ref="failure" minOccurs="0" maxOccurs="unbounded"/>
                <xs:element ref="system-out" minOccurs="0" maxOccurs="unbounded"/>
                <xs:element ref="system-err" minOccurs="0" maxOccurs="unbounded"/>
            </xs:sequence>
            <xs:attribute name="name" type="xs:string" use="required"/>
            <xs:attribute name="assertions" type="xs:string" use="optional"/>
            <xs:attribute name="time" type="xs:string" use="optional"/>
            <xs:attribute name="classname" type="xs:string" use="optional"/>
            <xs:attribute name="status" type="xs:string" use="optional"/>
        </xs:complexType>
    </xs:element>

    <xs:element name="testsuite">
        <xs:complexType>
            <xs:sequence>
                <xs:element ref="properties" minOccurs="0" maxOccurs="1"/>
                <xs:element ref="testcase" minOccurs="0" maxOccurs="unbounded"/>
                <xs:element ref="system-out" minOccurs="0" maxOccurs="1"/>
                <xs:element ref="system-err" minOccurs="0" maxOccurs="1"/>
            </xs:sequence>
            <xs:attribute name="name" type="xs:string" use="required"/>
            <xs:attribute name="tests" type="xs:string" use="required"/>
            <xs:attribute name="failures" type="xs:string" use="optional"/>
            <xs:attribute name="errors" type="xs:string" use="optional"/>
            <xs:attribute name="time" type="xs:string" use="optional"/>
            <xs:attribute name="disabled" type="xs:string" use="optional"/>
            <xs:attribute name="skipped" type="xs:string" use="optional"/>
            <xs:attribute name="timestamp" type="xs:string" use="optional"/>
            <xs:attribute name="hostname" type="xs:string" use="optional"/>
            <xs:attribute name="id" type="xs:string" use="optional"/>
            <xs:attribute name="package" type="xs:string" use="optional"/>
        </xs:complexType>
    </xs:element>

    <xs:element name="testsuites">
        <xs:complexType>
            <xs:sequence>
                <xs:element ref="testsuite" minOccurs="0" maxOccurs="unbounded"/>
            </xs:sequence>
            <xs:attribute name="name" type="xs:string" use="optional"/>
            <xs:attribute name="time" type="xs:string" use="optional"/>
            <xs:attribute name="tests" type="xs:string" use="optional"/>
            <xs:attribute name="failures" type="xs:string" use="optional"/>
            <xs:attribute name="disabled" type="xs:string" use="optional"/>
            <xs:attribute name="errors" type="xs:string" use="optional"/>
</xs:complexT
```

Today, in Inspec, it seems that : 

"failures" is called "failed"
"hostname" is called "target" and is placed inside each "testcase" instead of once inside "testsuite"
"skipped" is not present
"errors" is not present
"time" is missing inside "testsuite"

and others things i've not seen yet ^^

### 🌍 InSpec and Platform Version

```
$ inspec --version
2.1.59

$ cat /etc/os-release 
NAME="Ubuntu"
VERSION="16.04.4 LTS (Xenial Xerus)"
ID=ubuntu
ID_LIKE=debian
PRETTY_NAME="Ubuntu 16.04.4 LTS"
VERSION_ID="16.04"
HOME_URL="http://www.ubuntu.com/"
SUPPORT_URL="http://help.ubuntu.com/"
BUG_REPORT_URL="http://bugs.launchpad.net/ubuntu/"
VERSION_CODENAME=xenial
UBUNTU_CODENAME=xenial
```

### 🤔 Replication Case

```
$ inspec exec https://github.com/learn-chef/auditd/releases/download/v0.1.0/auditd-0.1.0.tar.gz --reporter=junit
[2018-04-30T14:24:07+02:00] WARN: Unrecognized content type: application/octet-stream. Assuming tar.gz
<?xml version='1.0'?>
<testsuites>
  <testsuite name='auditd' tests='1' failed='1'>
    <testcase name='System Package auditd should be installed' classname='auditd.Anonymous' target='local://' time='0.046789464'>
      <failure message='expected that `System Package auditd` is installed'/>
    </testcase>
  </testsuite>
</testsuites>
```

### 💁 Possible Solutions

Apply the XSD inside description :)

Regards,

Aha! Link: https://chef.aha.io/features/SH-2252

**Comment by clintoncwolfe** [comment]
The additions seem straightforward.

The replacements - failures for failed and target for hostname - are a bit more delicate.  It's likely users are relying on the existing (improper) behavior.  I'd suggest adding failed and keeping failures for the time being while we deprecate it.

A decision would also be needed around hostname / target.  `target` is the term for the object of the inspec run - which may be a host, a ddocker container, a cloud provider.  We could place that information in the `hostname` attribute, with the understanding that it may or may not be an actual host.

**Comment by clintoncwolfe** [comment]
https://github.com/windyroad/JUnit-Schema/blob/master/JUnit.xsd Seems to cover the entire schema, and addresses each of your concerns. Now that reporters are pluggable, I'm working on an improved junit2 reporter.

**Comment by clintoncwolfe** [comment]
The `junit2` reporter is now available, merged in #5085. I'm going to close this, thanks!
